src/optimistic-oracle/optimistic-oracle.js

```diff
@@ -0,0 +1,264 @@
+  Logger: logger,
+  createReferencePriceFeedForFinancialContract,
+} = require('@uma/financial-templates-lib');
+
+// load agent configuration
+const {
+  umaEverestId: UMA_EVEREST_ID,
+  optimisticOracle: optimisticOracleConfig,
+} = require('../../agent-config.json');
+
+const {
+  disputePriceErrorPercent,
+  cryptowatchApiKey: CRYPTOWATCH_API_KEY,
+  defipulseApiKey: DEFIPULSE_API_KEY,
+  tradermadeApiKey: TRADERMADE_API_KEY,
+  cmcApiKey: CMC_API_KEY,
+} = optimisticOracleConfig;
+
+const CHAIN_ID = 1; // mainnet
+
+// stores the optimistic oracle contract address
+let optimisticOracleAddress;
+
+logger.silent = true;
+
+// initialize global constants, web3 gets populated on initialization
+const getTime = () => Math.round(new Date().getTime() / 1000);
+const web3 = new Web3(new Web3.providers.HttpProvider(getJsonRpcUrl()));
+
+// create ethers interface object
+const optimisticOracleAbi = getAbi('OptimisticOracle');
+const iface = new ethers.utils.Interface(optimisticOracleAbi);
+
+/*
+Events we are interested in:
+
+    event RequestPrice(
+        address indexed requester,
+        bytes32 identifier,
+        uint256 timestamp,
+        bytes ancillaryData,
+        address currency,
+        uint256 reward,
+        uint256 finalFee
+    );
+
+    event ProposePrice(
+        address indexed requester,
+        address indexed proposer,
+        bytes32 identifier,
+        uint256 timestamp,
+        bytes ancillaryData,
+        int256 proposedPrice,
+        uint256 expirationTimestamp,
+        address currency
+    );
+*/
+const eventNames = ['RequestPrice', 'ProposePrice'];
+
+// calculate the % difference between 2 BigNumber values
+function calculatePercentError(first, second) {
+  const delta = first.minus(second).absoluteValue();
+  return delta.div(first).multipliedBy(100);
+}
+
+// creates a price feed using the UMA library
+async function createPriceFeed({ identifier, config }) {
+  // make a copy so we can change config w/o affecting the function argument (no-param-reassign)
+  const localConfig = config;
+
+  // try to create a price feed
+  // this typically fails if the target asset requires a specific API key,
+  // or the lookback value has not been set
+  let priceFeed = await createReferencePriceFeedForFinancialContract(
+    logger,
+    web3,
+    new Networker(),
+    getTime,
+    undefined, // no address needed since we're passing identifier explicitly
+    localConfig,
+    identifier,
+  ).catch();
+
+  // if the first attempt failed, set a lookback value and try again
+  if (!priceFeed) {
+    localConfig.lookback = 0;
+    priceFeed = await createReferencePriceFeedForFinancialContract(
+      logger,
+      web3,
+      new Networker(),
+      getTime,
+      undefined, // no address needed since we're passing identifier explicitly
+      localConfig,
+      identifier,
+    ).catch();
+  }
+
+  if (!priceFeed) {
+    throw Error(`Unable to create price feed for identifier '${identifier}'`);
```

Comment:
Should this use the `new` operator?
```suggestion
    throw new Error(`Unable to create price feed for identifier '${identifier}'`);
```